// @mui/x-charts/esm/internals/plugins/featurePlugins/useChartInteraction/useChartInteraction.js
import _extends from "@babel/runtime/helpers/esm/extends";
import useEventCallback from '@mui/utils/useEventCallback';
export const useChartInteraction = ({
  store
}) => {
  const cleanInteraction = useEventCallback(function cleanInteraction() {
    store.update({
      interaction: _extends({}, store.state.interaction, {
        pointer: null
      })
    });
  });
  const setLastUpdateSource = useEventCallback(function setLastUpdateSource(interaction) {
    if (store.state.interaction.lastUpdate !== interaction) {
      store.set('interaction', _extends({}, store.state.interaction, {
        lastUpdate: interaction
      }));
    }
  });
  const setPointerCoordinate = useEventCallback(function setPointerCoordinate(coordinate) {
    store.set('interaction', _extends({}, store.state.interaction, {
      pointer: coordinate,
      lastUpdate: coordinate !== null ? 'pointer' : store.state.interaction.lastUpdate
    }));
  });
  return {
    instance: {
      cleanInteraction,
      setLastUpdateSource,
      setPointerCoordinate
    }
  };
};
useChartInteraction.getInitialState = () => ({
  interaction: {
    item: null,
    pointer: null,
    lastUpdate: 'pointer'
  }
});
useChartInteraction.params = {};
